File: 2019 PAC II - AED - 0800 - Semana de Ejercicios - Ejercicio 2/JavaScript/Directory.js
function Directory(){
    this.numericalTree = new BST();
    this.alphabeticalTree = new BST();

    this.mainMenu = DirectoryMainMenu;
    this.addPrompt = DirectoryAddPrompt;
    this.searchPrompt = DirectorySearchPrompt;
    this.deletePrompt = DirectoryDeletePrompt;
    this.printPrompt = DirectoryPrintPrompt;
}

    function DirectoryMainMenu(){
        
        this.continue = true;
        menu = "Directorio de Contactos\n\nSeleccione una opción:\n1.Agregar Contacto\n2.Buscar Contacto\n3.Borrar Contacto\n4.Mostrar Contactos\n5.Salir\n";

        while(this.continue){
            option = prompt(menu);

            switch(option){
                case "1":
                    this.addPrompt();
                    break;

                case "2":
                    this.searchPrompt();
                    break;

                case "3":
                    this.deletePrompt();
                    break;

                case "4":
                    this.printPrompt();
                    break;

                case "5":
                    document.write("Programa Finalizado.");
                    this.continue = false;
                    break;

                default:
                    mensaje = `La opción ingresada ${option} no está en el menú.`;
                    alert(mensaje);
                    break;
            }
        }

    }

    function DirectoryAddPrompt(){
        name = prompt("Ingrese el nombre:");
        number = prompt("Ingrese el número:");

        wasAddedNumerically = this.numericalTree.addNumerically(new Node(new Contact(name, number)));
        wasAddedAlphabetically = this.alphabeticalTree.addAlphabetically(new Node(new Contact(name, number)));

        if(wasAddedNumerically && wasAddedAlphabetically){
            alert(`El contacto ${name}: ${number} fue agregado exitosamente.`);
        }
        else{
            alert(`El contacto ${name}: ${number} ya existe`);
        }
    }

    function DirectorySearchPrompt(){
        wasFound = null;
        searchBy = prompt("Desea buscar el contacto por:\n1.Nombre\n2.Número");
        if(searchBy == "1"){
            searchValue = prompt("Ingrese el nombre:");
            wasFound = this.alphabeticalTree.searchAlphabetically(searchValue);
        }
        else{
            searchValue = prompt("Ingrese el número:");
            wasFound = this.numericalTree.searchNumerically(searchValue);
        }

        if(wasFound){
            alert(`${wasFound.value.name}: ${wasFound.value.number}`);
        }
        else{
            alert(`El contacto no existe en el directorio.`);
        }
    }

    function DirectoryDeletePrompt(){
        wasDeleted = null;
        deleteBy = prompt("Desea borrar el contacto por:\n1.Nombre\n2.Número");
        if(deleteBy == "1"){
            deleteValue = prompt("Ingrese el nombre:");
            wasDeleted = this.alphabeticalTree.deleteAlphabetically(deleteValue);
        }
        else{
            deleteValue = prompt("Ingrese el número:");
            wasDeleted = this.numericalTree.deleteNumerically(deleteValue);
        }

        if(wasDeleted){
            alert(`El contacto deseado fue borrado.`);
        }
        else{
            alert(`El contacto no existe en el directorio.`);
        }

    }

    function DirectoryPrintPrompt(){
        printFormat = prompt("Desea verlo por:\n1.Orden Alfabético\n2.Orden Numerico");
        var tree = "";

        if(printFormat == "1"){
            tree = this.alphabeticalTree.print();
        }
        else{
            tree = this.numericalTree.print();
        }

        if(tree == ""){
            alert("El directorio está vacio.");
        }
        else{
            alert(tree);
        }
    }

    

File: 2019 PAC II - AED - 0800 - Semana de Ejercicios - Ejercicio 2/Python/Directory.py
from BST import*
from Contact import*

class Directory:
    def __init__(self):
        self.numericalTree = BST()
        self.alphabeticalTree = BST()

    def DirectoryAddinput(self):
        name = input("Ingrese el nombre:")
        number = input("Ingrese el número:")

        wasAddedNumerically = self.numericalTree.addNumerically(Node(Contact(name,number)))
        wasAddedAlphabetically = self.alphabeticalTree.addAlphabetically(Node(Contact(name,number)))

        if(wasAddedNumerically and wasAddedAlphabetically):
            input("El contacto %s: %s fue agregado exitosamente."%(name,number))
        
        else:
            input("El contacto %s: %s ya existe"%(name,number))
    
    
    def mainMenu(self):
       
        menu = """Directorio de Contactos:
        1.Agregar Contacto
        2.Buscar Contacto
        3.Borrar Contacto
        4.Mostrar Contactos
        5.Salir
        """

        Keep = True
        while(Keep):

            print("\n" + menu)

            eleccion = input("Elige: ")
            if eleccion is "1":
                print("Insertar")


            elif eleccion is "2":
                print("Ver")


            elif eleccion is "3":
                print("Actualizar")


            elif eleccion is "4":
                print("Eliminar")


            elif eleccion is "5":
                print("Salir")
                Keep = False
            else:
                #Equivalente a 'default'
                print("Ninguna opción válida seleccionada")


Prueba = Directory()
Prueba.mainMenu()